cpp/src/datetime/datetime_ops.cu

```diff
@@ -648,6 +628,26 @@ std::unique_ptr<column> extract_second(column_view const& column, rmm::device_as
   return detail::extract_second(column, cudf::get_default_stream(), mr);
 }
 
+std::unique_ptr<cudf::column> extract_datetime_component(cudf::column_view const& column,
+                                                         datetime_component component,
+                                                         rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  switch (component) {
+    case datetime_component::YEAR: return extract_year(column, mr);
```

Comment:
If we plan to deprecate functions like `extract_year`, we should reverse the call stack here.

This function should dispatch to `return detail::apply_datetime_op<detail::extract_component_operator<...>, cudf::type_id::INT16>(column, stream, mr);` based on the enum, and we should make `extract_year` call `extract_datetime_component(column, datetime_component::YEAR, mr);`.

Reply:
Made these changes with the latest commits - let me know if this is along the lines of what you meant.